makefiles/features_check.inc.mk

```diff
@@ -0,0 +1,38 @@
+# Check if all required FEATURES are provided
+
+FEATURES_OPTIONAL_ONLY := $(sort $(filter-out $(FEATURES_REQUIRED),$(FEATURES_OPTIONAL)))
+FEATURES_OPTIONAL_USED := $(sort $(filter $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
+# Optional features that will not be used because they are not provided
+FEATURES_OPTIONAL_MISSING := $(sort $(filter-out $(FEATURES_PROVIDED),$(FEATURES_OPTIONAL_ONLY)))
+
+# Features that are used without taking "one out of" dependencies into account
+FEATURES_USED_SO_FAR := $(sort $(FEATURES_REQUIRED) $(FEATURES_OPTIONAL_USED))
+
+# Features that are provided and not blacklisted
+FEATURES_USABLE := $(filter-out $(FEATURES_BLACKLIST),$(FEATURES_PROVIDED))
+
+# Additionally required features due to the "one out of" dependencies
+FEATURES_REQUIRED_ONE_OUT_OF := $(foreach item,$(FEATURES_REQUIRED_ANY),$(word 1,$(filter $(subst |, ,$(item)),$(FEATURES_USED_SO_FAR) $(FEATURES_USABLE)) $(item)))
```

Comment:
Was in on purpose do remove all the intermediate functions here?

Reply:
The approach changed. The previous implementation used indeed functions that either added nothing in case one of the required features in an item was already used, or added one of requested features.

The new approach does not use `$(call)` at all but always extends the list; either with already included features (which will be de-duplicated later on), or one of the features requested and provided.

It turned out that this approach performed better than `$(call )`.

Reply:
Ok! Coud you still re-add some more description of the behavior (e.g. that it will add the first match, etc., basically what is pertinent from the the old description)

Reply:
resolved